### DISCUSSION TITLE
Would there be any merit in rewriting Xournal++ in Rust?

### DISCUSSION THREAD

**Discussion by K4LCIFER** [xournalpp/xournalpp#3667] [General]
There is already an existing movement of people that are rewriting many existing programas which are originally written in usually C or C++ in Rust. Xournal++ is already a C++ rewrite of Xournal which was in C, so I'm curious if there has been any talk about rewriting Xournal++ in Rust. If so, I would be curious as to the reasoning behind making such a decision.

**Comment by Technius** [comment]
Some of the maintainers expressed some interest in a Rust rewrite, but that has not really gone anywhere outside of the planning stage. Personally, I don't think it's worth the effort to port to Rust, as that would mean rewriting much of the features, bug fixes, and infrastructure that we set up over the years.
